src/main/java/com/aol/cyclops/control/Maybe.java

```diff
@@ -72,591 +73,674 @@ public void odd() {
-
-    /* (non-Javadoc)
-     * @see com.aol.cyclops.types.Zippable#zip(org.jooq.lambda.Seq, java.util.function.BiFunction)
-     */
-    @Override
-    default <U, R> Maybe<R> zip(final Seq<? extends U> other, final BiFunction<? super T, ? super U, ? extends R> zipper) {
-        return (Maybe<R>) MonadicValue1.super.zip(other, zipper);
-    }
-
-    /* (non-Javadoc)
-     * @see com.aol.cyclops.types.Zippable#zip(java.util.stream.Stream, java.util.function.BiFunction)
-     */
-    @Override
-    default <U, R> Maybe<R> zip(final Stream<? extends U> other, final BiFunction<? super T, ? super U, ? extends R> zipper) {
-
-        return (Maybe<R>) MonadicValue1.super.zip(other, zipper);
-    }
-
-    /* (non-Javadoc)
-     * @see com.aol.cyclops.types.Zippable#zip(java.util.stream.Stream)
-     */
-    @Override
-    default <U> Maybe<Tuple2<T, U>> zip(final Stream<? extends U> other) {
-
-        return (Maybe) MonadicValue1.super.zip(other);
-    }
-
-    /* (non-Javadoc)
-     * @see com.aol.cyclops.types.Zippable#zip(org.jooq.lambda.Seq)
-     */
-    @Override
-    default <U> Maybe<Tuple2<T, U>> zip(final Seq<? extends U> other) {
-
-        return (Maybe) MonadicValue1.super.zip(other);
-    }
-
-    /* (non-Javadoc)
-     * @see com.aol.cyclops.types.Zippable#zip(java.lang.Iterable)
-     */
-    @Override
-    default <U> Maybe<Tuple2<T, U>> zip(final Iterable<? extends U> other) {
-
-        return (Maybe) MonadicValue1.super.zip(other);
-    }
-
-    @Override
-    default <T> Maybe<T> unit(final T unit) {
-        return Maybe.of(unit);
-    }
-
-    /* (non-Javadoc)
-     * @see com.aol.cyclops.types.MonadicValue#coflatMap(java.util.function.Function)
-     */
-    @Override
-    default <R> Maybe<R> coflatMap(final Function<? super MonadicValue<T>, R> mapper) {
-        return (Maybe<R>) MonadicValue1.super.coflatMap(mapper);
-    }
-
-    /* cojoin
-     * (non-Javadoc)
-     * @see com.aol.cyclops.types.MonadicValue#nest()
-     */
-    @Override
-    default Maybe<MonadicValue<T>> nest() {
-        return (Maybe<MonadicValue<T>>) MonadicValue1.super.nest();
-    }
-
-    /* (non-Javadoc)
-     * @see com.aol.cyclops.types.MonadicValue2#combine(com.aol.cyclops.Monoid, com.aol.cyclops.types.MonadicValue2)
-     */
-    @Override
-    default Maybe<T> combineEager(final Monoid<T> monoid, final MonadicValue<? extends T> v2) {
-        return (Maybe<T>) MonadicValue1.super.combineEager(monoid, v2);
-    }
-
-    /* (non-Javadoc)
-     * @see com.aol.cyclops.value.Value#toMaybe()
-     */
-    @Override
-    default Maybe<T> toMaybe() {
-        return this;
-    }
-
-    @Override
+		extends MonadicValue1<T>, Supplier<T>, ConvertableFunctor<T>, Filterable<T>, ApplicativeFunctor<T>, Matchable.ValueAndOptionalMatcher<T> {
+
+	final static Maybe EMPTY = new Nothing<>();
+
+	static <T> Maybe<T> none() {
+		return EMPTY;
+	}
+
+	@Override
+	default <R> Maybe<R> flatMapIterable(Function<? super T, ? extends Iterable<? extends R>> mapper) {
```

Comment:
Same comment about Javadoc (I'm working on getting all these methods to have Javadoc and then on the core types will add in more detailed documentation there too).